### DISCUSSION TITLE
How is curl searching for the cert?

### DISCUSSION THREAD

**Discussion by Dees7** [curl/curl#9462] [Q&A] [answered]
Hello. 
Please, can someone explain? (I don't have skills to understand it from source)

I has some trouble with site (ssl error).
When I run
`strace curl -Is https://www.thevista.ru `
I see in log
`stat("/etc/ssl/certs/16744f0c.0", 0x7fff9fbca0c0) = -1 ENOENT (No such file or directory)`

How curl calculates 16744f0c from certs on site (cn=www.thevista.ru) to check certificate chain?
for example how calculate for [Let's Encrypt Authority X3](https://letsencrypt.org/certificates/)?

PS: I put cert  CN=AlphaSSL CA with name 16744f0c.0 and now all ok. But I'd like to understand how curl calculates.

**Comment by dfandrich** [comment, marked as answer]
On Fri, Sep 09, 2022 at 11:35:58AM -0700, Dees7 wrote:
> How curl calculates 16744f0c from certs on site (cn=www.thevista.ru) to check
> certificate chain?

This isn't done by curl, but rather by the TLS library that curl is 
using. "curl -V" will tell you that.

> PS: I put cert CN=AlphaSSL CA with name 16744f0c.0 and now all ok. But I'd like
> to understand how curl calculates.

You'll have to look at documentation for the TLS library. For example, 
OpenSSL has a utility c_rehash that generates hashed links for its 
certificates. If you're interested to see how it calculates those 
hashes, you could look at its source code. But, this is really an 
internal detail of how it operates.  Just use c_rehash (or whatever 
utility is supplied by your TLS library) to generate the links and let 
the library do its thing.

**Reply by Dees7** [reply]
Grate! thanks! )
There is the [more info](https://unix.stackexchange.com/questions/698814/why-are-so-many-strange-certificate-filenames-ending-with-0-in-etc-ssl-certs)
